plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -190,19 +191,33 @@ func defaultPredicates() sets.String {
 			},
 		),
 	)
+}
+
+// ApplyFeatures applys algorithm by feature gates.
+func ApplyFeatures() {
```

Comment:
I am not sure about the name of this function and I think separating this from `defaultPredicates` reduces readability. Now, one could get wrong impression about the set of default predicates by looking at `defaultPredicates` function.

Reply:
Fair point :). Here's the option I used to try:

1. Register default predicates in `init()`, and use `ApplyFeatures` to update/apply predicates by features gate
1. Move the logic in `init()` to a new func, named `Install()`; and call `Install()` in both `init()` and `app.Run`
1. Similar to /2, but call `Install()` explicitly if necessary
1. Update getFitPredicateFunctions to return predicates by feature gate

The PR follows option 1 :). The second option requires `Install()` can be re-run, and the result dependent on feature gates,
the function user maybe confused with the result. The drawback of option 3 is risk, if we missed a `Install()`, some feature
may not work.

Reply:
I am a bit confused. I thought 'defaultPredicates()' which was being called from `init()` was already registering the right predicate functions based on feature gates. So, why is the specific change that this PR make in registering predicates?

Reply:
The `init()` will be executed when package imported; the feature gate is not updated by args at that time.

Reply:
I see. I would call this function something like "ApplyFeatureGates" or something like that to make it more clear what it does.